chain/near/src/adapter.rs

```diff
@@ -25,10 +36,71 @@ impl bc::TriggerFilter<Chain> for TriggerFilter {
     }
 
     fn to_firehose_filter(self) -> Vec<prost_types::Any> {
-        vec![]
+        let TriggerFilter {
+            block_filter: block,
+            receipt_filter: receipt,
+        } = self;
+
+        if block.trigger_every_block {
+            return vec![];
+        }
+
+        if receipt.is_empty() {
+            return vec![];
+        }
+
+        let filter = BasicReceiptFilter {
+            accounts: receipt.accounts.into_iter().collect(),
+        };
+
+        vec![Any {
+            type_url: BASIC_RECEIPT_FILTER_TYPE_URL.into(),
+            value: filter.encode_to_vec(),
+        }]
+    }
+}
+
+pub(crate) type Account = String;
+
+/// NearReceiptFilter requires the account to be set, it will match every receipt where `source.account` is the recipient.
+/// see docs: https://thegraph.com/docs/en/supported-networks/near/
+#[derive(Clone, Debug, Default)]
+pub(crate) struct NearReceiptFilter {
+    pub accounts: HashSet<Account>,
+}
+
+impl NearReceiptFilter {
+    pub fn matches(&self, account: &String) -> bool {
+        !self.is_empty() && self.accounts.contains(account)
+    }
+
+    pub fn is_empty(&self) -> bool {
+        let NearReceiptFilter { accounts } = self;
+
+        accounts.is_empty()
+    }
+
+    pub fn from_data_sources<'a>(iter: impl IntoIterator<Item = &'a DataSource>) -> Self {
+        iter.into_iter()
+            .filter(|data_source| {
+                data_source.source.account.is_some()
+                    && !data_source.mapping.receipt_handlers.is_empty()
+            })
+            .fold(Self::default(), |mut filter_opt, data_source| {
```

Comment:
This is more of a `collect` than a `fold`.

Reply:
Not sure I follow what you mean here.

Reply:
do you mean map then collect and then .extend on the hashset?

Reply:
Not even extend, just initialize `accounts` to the collect.

Reply:
ah yes of course! 🤦